core/src/main/java/ai/timefold/solver/core/impl/domain/variable/declarative/ChangedVariableNotifier.java

```diff
@@ -17,6 +21,17 @@ public record ChangedVariableNotifier<Solution_>(BiConsumer<VariableDescriptor<S
             },
             null);
 
+    public CollectionInverseVariableSupply getCollectionInverseVariableSupply(VariableMetaModel<?, ?, ?> variableMetaModel) {
+        if (innerScoreDirector == null) {
+            return entity -> Collections.emptyList();
+        } else {
+            var solutionDescriptor = innerScoreDirector.getSolutionDescriptor();
+            var variableDescriptor = solutionDescriptor.getEntityDescriptorStrict(variableMetaModel.entity().type())
+                    .getVariableDescriptor(variableMetaModel.name());
+            return innerScoreDirector.getSupplyManager().demand(new CollectionInverseVariableDemand<>(variableDescriptor));
```

Comment:
Is it possible for the method to be called multiple times with the same variable descriptor? This method should, under no circumstances, cause the demands to accumulate.

Reply:
Yes; but wouldn't the demands share the same supply since the variable descriptor the same instance? The demands are also Solver lifecycle long: https://github.com/TimefoldAI/timefold-solver/blob/16e2464fda033694b962d06e1c6895e1e838a74b/core/src/main/java/ai/timefold/solver/core/impl/domain/variable/supply/AbstractVariableDescriptorBasedDemand.java#L25-L40

Reply:
Yes, the demands would result in the same supply. At the same time though, a counter would be increased; so this effectively creates "a leak", a supply that will never be un-demanded.

Is it a problem? Probably not.